src/gfx/linalg.hpp

```diff
@@ -14,6 +15,66 @@ inline void packFloat4x4(const float4x4 &mat, float *dst) {
   }
 }
 
+// compute the near and far intersections of the cube (stored in the x and y components) using the slab method
+// no intersection means vec.x > vec.y (really tNear > tFar)
+inline float2 intersectAABB(const float3 &rayOrigin, const float3 &rayDir, float3 boxMin, float3 boxMax) {
+  // https://gist.github.com/DomNomNom/46bb1ce47f68d255fd5d
+  float3 tMin = (boxMin - rayOrigin) / rayDir;
+  float3 tMax = (boxMax - rayOrigin) / rayDir;
+  float3 t1 = linalg::min(tMin, tMax);
+  float3 t2 = linalg::max(tMin, tMax);
+  float tNear = linalg::max(linalg::max(t1.x, t1.y), t1.z);
+  float tFar = linalg::min(linalg::min(t2.x, t2.y), t2.z);
+  return float2(tNear, tFar);
+};
+
+// Compute intersection of a ray with a plane
+// Returns distance in t
```

Comment:
t is not a parameter in the argument list nor a unit. If the intention is to return outDistance in the range of [0, 1] explain that. Or if t can have values above 1 explain that outDistance is the distance between rayOrigin and the point of intersection.

Other formal parameters need to be explained too. planePoint - any point which lies on the plane. The user of the function must be able to determine how it works from the signature.

Reply:
Fixed comment